common/network-common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java

```diff
@@ -171,6 +186,21 @@ public void handle(ResponseMessage message) throws Exception {
         listener.onSuccess(resp.streamChunkId.chunkIndex, resp.body());
         resp.body().release();
       }
+      // The response is `ChunkFetchSuccess`. It's either because the request was a normal chunk
+      // fetch request, or the server side is an old version that doesn't support fetch chunk as
+      // stream. So the next line is either a no-op, or remove the callback that will never be
+      // called later.
+      outstandingFetchAsStreams.remove(resp.streamChunkId);
+    } else if (message instanceof ChunkFetchStreamResponse) {
+      ChunkFetchStreamResponse resp = (ChunkFetchStreamResponse) message;
+      StreamCallback callback = outstandingFetchAsStreams.get(resp.streamChunkId);
```

Comment:
Also remove this `callback` from `outstandingFetchAsStreams` ?

Reply:
good catch!